src/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -149,8 +143,12 @@ create_synthetic_image_from_background_and_starter(const flex_image &starter,
     log_and_throw("Background image is not decoded into raw format.");
   }
 
-  boost::gil::rgba8_image_t::view_t starter_image_view =
-      create_starter_image_view(starter);
+  flex_image rgba_flex_image = create_rgba_flex_image(starter);
+  boost::gil::rgba8_image_t::view_t starter_image_view = interleaved_view(
+      rgba_flex_image.m_width, rgba_flex_image.m_height,
+      (boost::gil::rgba8_pixel_t *)(rgba_flex_image.get_image_data()),
```

Comment:
Ah, yes, you cannot have a function that safely returns a view to a local variable inside that function.

While we're at it, could you replace this C-style cast with `reinterpret_cast<boost::gil::rgba8_pixel_t *>`?

Reply:
Fixed.